src/Account/Account.js

```diff
@@ -103,14 +104,21 @@ class Account {
     this.store = wallet.storage.store;
     this.storage = wallet.storage;
 
-    if (this.type === WALLET_TYPES.HDWALLET) {
+    if (this.walletType === WALLET_TYPES.HDWALLET) {
       this.storage.importAccounts({
         label: this.label,
         path: this.BIP44PATH,
         network: this.network,
       }, this.walletId);
     }
-    if (this.type === WALLET_TYPES.SINGLE_ADDRESS) {
+    if (this.walletType === WALLET_TYPES.HDEXTPUBLIC) {
+      this.storage.importSingleAddress({
```

Comment:
I think if we use this/self, it should be decomposed in the beginning of the function, so dependencies are more transparent.